sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-supplier.service.ts

```diff
@@ -0,0 +1,186 @@
+          summary.error("Unexpected error.", error);
+        }
+      } else {
+        summary.info("No supplier found on CAS.");
+      }
+    }
+    return suppliersUpdated;
+  }
+
+  /**
+   * Updates CAS supplier table.
+   * @param casSupplier CAS supplier to be updated.
+   * @param casSupplierResponseItem CAS supplier response item from CAS request.
+   * @param supplierStatus CAS supplier status to be updated.
+   * @returns update result.
+   */
+  async updateCASSupplier(
+    casSupplier: CASSupplier,
+    casSupplierResponseItem: CASSupplierResponseItem,
+    supplierStatus: SupplierStatus,
+  ): Promise<UpdateResult> {
+    // When multiple exists, only the active one should be saved.
+    // We will not be saving the array received at this moment, only a single entry from the received list should be persisted as JSONB.
+    const activeSupplierAddress = casSupplierResponseItem.supplieraddress.find(
+      (address) => address.status === CAS_SUPPLIER_ADDRESS_ACTIVE_STATUS,
+    );
+    let supplierAddressToUpdate: SupplierAddress = null;
+    if (activeSupplierAddress) {
+      supplierAddressToUpdate = {
+        supplierSiteCode: activeSupplierAddress.suppliersitecode,
+        addressLine1: activeSupplierAddress.addressline1,
+        city: activeSupplierAddress.city,
+        provinceState: activeSupplierAddress.province,
+        country: activeSupplierAddress.country,
+        postalCode: activeSupplierAddress.postalcode,
+        status: activeSupplierAddress.status,
+        siteProtected: activeSupplierAddress.siteprotected,
+        lastUpdated: new Date(activeSupplierAddress.lastupdated),
+      };
+    }
+    const now = new Date();
+    const systemUser = this.systemUsersService.systemUser;
+    return this.casSupplierRepo.update(
+      {
+        id: casSupplier.id,
+      },
+      {
+        supplierNumber: casSupplierResponseItem.suppliernumber,
+        supplierName: casSupplierResponseItem.suppliername,
+        status: casSupplierResponseItem.status,
+        supplierProtected: casSupplierResponseItem.supplierprotected === "Y",
+        lastUpdated: new Date(casSupplierResponseItem.lastupdated),
+        supplierAddress: supplierAddressToUpdate,
+        supplierStatus,
+        supplierStatusUpdatedOn: now,
+        isValid: true,
+        updatedAt: now,
+        modifier: systemUser,
+      },
+    );
+  }
+
+  /**
+   * Gets a list of CAS suppliers to be updated from CAS supplier table.
+   * @returns a list of CAS suppliers to be updated.
+   */
+  async getStudentsToUpdateSupplierInformation(): Promise<CASSupplier[]> {
+    return this.casSupplierRepo.find({
+      select: {
+        id: true,
+        student: {
+          id: true,
+          sinValidation: { sin: true, isValidSIN: true },
+          user: { lastName: true },
+        },
+      },
+      relations: {
+        student: { sinValidation: true, user: true },
+      },
+      where: {
+        isValid: Not(true),
```

Comment:
Since `isValid` is not nullable this can be simplified as `isValid: false`, right?